koku/masu/api/trino.py

```diff
@@ -40,10 +40,12 @@ def trino_query(request):
         if schema_name is None:
             errmsg = "Must provide a schema key to run."
             return Response({"Error": errmsg}, status=status.HTTP_400_BAD_REQUEST)
+
         lowered_query = query.lower()
         dissallowed_keywords = ["delete", "insert", "update", "alter", "create", "drop", "grant"]
-        for keyword in dissallowed_keywords:
-            if keyword in lowered_query:
+
+        for keyword in sqlparse.parse(lowered_query)[0].flatten():
+            if keyword.value.lower() in dissallowed_keywords:
                 errmsg = f"This endpoint does not allow a {keyword} operation to be performed."
                 return Response({"Error": errmsg}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
I don't know the particular benefit of using sqlparse. This could have been handled by splitting the query on spaces and just comparing the lists via a set operation:
```
        lowered_query = set(query.lower().split(" "))
        dissallowed_keywords = {"delete", "insert", "update", "alter", "create", "drop", "grant"}
        if keywords := dissallowed_keywords.intersection(lowered_query):
            errmsg = f"This endpoint does not allow a {keywords} operation to be performed."
            return Response({"Error": errmsg}, status=status.HTTP_400_BAD_REQUEST)
```

Nothing wrong with using sqlparse, it just seems overkill to pull in a library for this.

Reply:
thanks for the suggestion 👍🏾